3-platform/warehouse/spec/src/main/java/com/_6core/platform/warehousespec/rest/v1/InventoryManagementApi.java

```diff
@@ -0,0 +1,90 @@
+package com._6core.platform.warehousespec.rest.v1;
+
+import com._6core.platform.warehousespec.rest.v1.dto.inventory.InventoryAllocateRequest;
+import com._6core.platform.warehousespec.rest.v1.dto.inventory.InventoryAllocateResponse;
+import com._6core.platform.warehousespec.rest.v1.dto.inventory.InventorySupplyRequest;
+import com._6core.platform.warehousespec.rest.v1.dto.inventory.InventorySupplyResponse;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import reactor.core.publisher.Mono;
+
+/**
+ * Interface for inventory management  in the Warehouse Service.
+ */
+
+@Tag(name = "Inventory management",
+        description = "REST specification for allocating the inventory and for supplying to the existing inventory")
+public interface InventoryManagementApi {
+
+    /**
+     * This API endpoint allows you to allocate a new space for the inventory in the warehouse.
+     *
+     *  - The response status code will be:
+     *      - 200 (OK) - if the inventory transfer is successful.
+     *      - 400 (Bad Request) - if there is a field validation error
+     *      - 401 (Unauthorized) - if there is an authentication error
+     *      - 404 (Not Found) - if the inventory item with the provided ID is not found.
+     *      - 500 (Internal Server Error) - if an unexpected error occurs during the transfer process.
+     */
+
+    @Operation(summary = "Allocate inventory", description = "Allocate a new space for the inventory in our warehouse")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK", content = {
+                    @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = InventoryAllocateResponse.class))}),
+            @ApiResponse(responseCode = "400", description = "Field validation error"),
+            @ApiResponse(responseCode = "401", description = "Authentication error"),
+            @ApiResponse(responseCode = "404", description = "inventory not found"),
+            @ApiResponse(responseCode = "500", description = "Internal server error")
+    })
+    @PostMapping("/inventories/{inventoryId}/allocations")
```

Comment:
REST typically provides the resources, e.g. inventories. This is why we name it in the plural.
Is `allocations` the resource? I don't think so. 

`Allocate` is verb and URI convention recommends getting rid of `verbs`. Why? We have methods like POST, PATCH, PUT ,... that say to us what endpoint is going to do. 
So adding the verb is redundant in this case. 

The same for other endpoints.
```suggestion
    @PostMapping("/inventories/{inventoryId}")
```